src/v2/Apps/Auction/Routes/__tests__/ConfirmBid.jest.tsx

```diff
@@ -639,6 +668,36 @@ describe("Routes/ConfirmBid", () => {
       })
     })
 
+    it("takes the user back to the /auction/:sale_slug/confirm-registration page when BIDDER_NOT_QUALIFIED is returned", async () => {
+      const env = setupTestEnv()
+      const page = await env.buildPage({
+        mockData: FixtureForUnregisteredUserWithCreditCard,
+      })
+      env.mutations.useResultsOnce(createBidderPositionWithBidderNotQualified)
+
+      await page.agreeToTerms()
+      await page.submitForm()
+
+      expect(window.location.assign).toHaveBeenCalledWith(
```

Comment:
nice - so if I understand correctly, auction/:sale_slug/confirm-registration in turn redirects to the sale overview page, with the registration pending displayed? Very elegant - much better user experience

Reply:
I actually thought this specific assertion with `window.location` was a smell... I initially tried to use [`found`'s router](https://github.com/4Catalyzer/found#programmatic-navigation) but it didn't work because the auction routes are not part of the main App Shell. Hopefully we'l be able to replace it soon.